actor-typed/src/main/mima-filters/1.0.x.backwards.excludes/issue-3413.excludes

```diff
@@ -0,0 +1,6 @@
+ProblemFilters.exclude[MissingFieldProblem]("org.apache.pekko.actor.typed.scaladsl.Behaviors.Supervise")
+ProblemFilters.exclude[MissingClassProblem]("org.apache.pekko.actor.typed.scaladsl.Behaviors$Supervise$")
+ProblemFilters.exclude[MissingClassProblem]("org.apache.pekko.actor.typed.scaladsl.Behaviors$Supervise")
+ProblemFilters.exclude[MissingClassProblem]("org.apache.pekko.actor.typed.javadsl.Behaviors$Supervise")
+ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.pekko.actor.typed.scaladsl.Behaviors.supervise")
+ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.pekko.actor.typed.javadsl.Behaviors.supervise")
```

Comment:
I really think we should not just add things to mima backwards excludes, but also add comments why the exclusions are justified.

In this case it looks, to me, like the mima errors were actually legitimate and this PR breaks our binary compatibility promises documented at https://pekko.apache.org/docs/pekko/current/common/binary-compatibility-rules.html and https://github.com/apache/pekko/blob/main/CONTRIBUTING.md#binary-compatibility

To make things explicit: suppose we have an application 'A' which uses a library 'L' which depends on Pekko 1.0.0 . The point of the binary compatibility rules is that it should be safe for 'A' to update to Pekko 1.1.0 without breaking.

In this case, if `L` contained:

```
val s: Behaviors.Supervise = Behaviors.supervise(SupervisedActor.create())
```

... AFAICS this code would throw a runtime exception when executed with any Pekko version that has the change in this PR, because `Behaviors.Supervise` no longer exists.

Reply:
> I really think we should not just add things to mima backwards excludes, but also add comments why the exclusions are justified.

I totally agree with that.

> In this case it looks, to me, like the mima errors were actually legitimate

Yes. 


What should we do when we face such API changes?

Reply:
> What should we do when we face such API changes?

It depends on the situation. I haven't looked closely, but in this case, it looks like the SuperviseBehaviour was introduced to reduce code duplication between the two `Behavior.Supervise` classes, is that correct? I wonder if we could restore binary compatibility while still avoiding code duplication by restoring the `Behavior.Supervise` classes, but having them `extends SuperviseBehaviour`. If that does not work, possibly we should just choose to accept some duplication to achieve compatibility.

Reply:
I has been fix these binary compatibility via some duplicate code. would you take a look?